### ISSUE TITLE
Incompatibility with connectivity plugin because of 'Reachability dependency/file' (iOS platform)

### ISSUE THREAD

**Issue by mpowloka** [chinloyal/pusher_client#6]
### Steps to reproduce

in pubspec.yaml decalre poth flutter_pusher and connectivity plugins.

What did you expect to happen?

iOS build will fail. (Xcode build will somehow pass, but app won't be installed)

What actually happened? Submit stack traces or anything that you think would help.

The problem is some file/dependency named Reachability.
Another pusher plugin suffers the same problem: https://github.com/HomeXLabs/pusher-websocket-flutter/issues/11

...

**Comment by mpowloka** [comment]
Any workaround or solution is welcomed. So far I didn't find a way to deal with this.

**Comment by chinloyal** [comment]
Yeah this is a common issue, the connectivity package is incompatible with because of the Reachability library for. The maintainer of the Reachability library needs to update their library to resolve this issue.

The workaround I found was to just use [data_connection_checker](https://pub.dev/packages/data_connection_checker) instead of connectivity, they both work the same way

**Comment by meteorSD** [comment]
Could you please explain how to implement your workaround ?

**Comment by chinloyal** [comment]
The package is pretty straightforward, simply use data_connection_checker package instead of connectivity package. They both work the same however data_connection_checker is compatible with this package and other pusher clients

**Comment by muhammadhassamriaz** [comment]
Any solution for this issue ?

**Comment by thethtun** [comment]
This is PITA! Have anyone found better ways to solve this?
